pkg/highlight/parser.go

```diff
@@ -7,6 +7,9 @@ import (
 	"regexp"
 
 	"gopkg.in/yaml.v2"
+
+	"github.com/zyedidia/micro/v2/internal/config"
+	"github.com/zyedidia/micro/v2/internal/screen"
```

Comment:
I guess originally the `pkg/highlight` package was supposed to be "independent" of the rest of micro (e.g. to be potentially "reusable" in completely different projects, that are just required to implement the same `LineStates` interface as micro).

I doubt we still need to adhere to that assumption and cannot just assume the highlighter to be a part of micro and nothing more. However, it seems we can still easily avoid importing these micro's internal packages here. Is `pkg/highlight/parser.go` really the best place to add these helper functions to?

So now I'm thinking: isn't it better to add them to `internal/buffer/buffer.go`, just before `UpdateRules()`, just as private functions? I'm thinking about that because inside `UpdateRules()` you still aren't completely getting rid of all those `f.Data()`, `highlight.ParseFile()`, `screen.TermMessage()` and so on, i.e. they just become scattered between `UpdateRules()` itself and these new helper functions. So maybe it's better to keep them located close to `UpdateRules()` itself?

Reply:
Ok, then we simply move it into the `buffer.go`. This change doesn't hurt, doesn't pollute the includes of `parser.go` and as you mentioned...it stays independent.